test: Cancel Mentorship Relation

mini-mini-milo opened the pull request:
Tested valid user cancelling valid task with missing authentication token
Tested valid user cancelling valid task with expired authentication token

### Description
Added tests for cancelling mentorship relations:
1. valid user cancelling valid task with missing authentication token (fail)
2. valid user cancelling valid task with expired authentication token (fail)
3. valid user cancelling when not involved in the relationship (fail)

Fixes #307 

### Type of Change:

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Ran locally using "python -m unittest discover tests"


### Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

Commit:
test: Cancel Mentorship Relation

Tested valid user cancelling valid task with missing authentication token
Tested valid user cancelling valid task with expired authentication token

Tlazypanda reviewed (changes requested):
Tests missing from this [table](https://github.com/systers/mentorship-backend/wiki/Quality-Assurance#cancel-relation).Can you add those tests?

isabelcosta reviewed (changes requested):
Hey @mini-mini-milo if this PR is part of the GCI competition, I believe one of the steps is that this PR should be targeting the `gci-dev` branch. You can change this by clicking the edit button next to the pull request title, and change base branch from `develop` to `gci-dev`.

Commit:
test: Add Cancel Mentorship Relation Tests

Adds a third user not involved in relationship
Tests third user cancelling relationship (fail)
Part of issue #307

isabelcosta commented on tests/mentorship_relation/test_api_cancel_relation.py:
```diff
@@ -61,7 +77,61 @@ def test__mentee_cancel_mentorship_relation(self):
+            self.assertDictEqual(messages.AUTHORISATION_TOKEN_IS_MISSING, 
+                                 json.loads(response.data))
+            
+    def test__mentee_cancel_token_missing(self):
```
This test and the test above are exactly equal, so this one can be deleted.
Can you please delete this?

isabelcosta commented on tests/mentorship_relation/test_api_cancel_relation.py:
```diff
@@ -61,7 +77,61 @@ def test__mentee_cancel_mentorship_relation(self):
+    
+    #missing authorisation token
+    
+    def test__mentor_cancel_token_missing(self):
```
```suggestion
    def test_mentor_cancel_token_missing(self):
```

isabelcosta replied:
I know there are other tests in this file with double `_`, it must have been a typo that stayed. we can fix it later.

isabelcosta commented on tests/mentorship_relation/test_api_cancel_relation.py:
```diff
@@ -61,7 +77,61 @@ def test__mentee_cancel_mentorship_relation(self):
+            
+    #expired authorisation token
+            
+    def test__mentor_cancel_token_expired(self):
```
```suggestion
    def test_mentor_cancel_token_expired(self):
```

isabelcosta commented on tests/mentorship_relation/test_api_cancel_relation.py:
```diff
@@ -61,7 +77,61 @@ def test__mentee_cancel_mentorship_relation(self):
+            self.assertEqual(MentorshipRelationState.ACCEPTED, self.mentorship_relation.state)
+            self.assertDictEqual(messages.TOKEN_HAS_EXPIRED, json.loads(response.data))
+            
+    def test__mentee_cancel_token_expired(self):
```
```suggestion
    def test_mentee_cancel_token_expired(self):
```

isabelcosta commented on tests/test_data.py:
```diff
@@ -23,3 +23,12 @@
+    "username": "user3",
+    "password": "user3_pwd",
+    "terms_and_conditions_checked": True
+}
```
Please add a blank line here in the end of the file

isabelcosta commented on tests/mentorship_relation/test_api_cancel_relation.py:
```diff
@@ -61,7 +77,61 @@ def test__mentee_cancel_mentorship_relation(self):
+        
+    #not involved in mentorship    
+        
+    def test__user_not_in_mentorship(self):
```
```suggestion
    def test_user_not_involved_tries_to_cancel_relation(self):
```

isabelcosta reviewed (changes requested):
Hey @mini-mini-milo I left some request changes. Its small things, you are almost done!
Thank you for your contribution 🙌

Commit:
test: Minor Changes to Cancel Relation

isabelcosta reviewed (changes requested):
You deleted one test that was ok, I just asked you to remove one of the tests which was `test__mentor_cancel_token_missing`.
You also removed `test__mentee_cancel_token_expired` test, that was needed. Can you please put that test again?

Commit:
test: Added mentee_cancel_token_expired

Tlazypanda reviewed (approved):
Approved :)

Commit:
feat: Improve Change Password Functionality

adds new error message
Changes password only when the new password does not match the current one

isabelcosta commented:
Thank you for your contribution here @mini-mini-milo  🎉 
I'm going to close this PR, since we decided to merge another one.
